src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -471,6 +472,17 @@ private IEnumerable<(ManifestId id, ManifestVersion version, SdkFeatureBand feat
                     ManifestVersion manifestVersion;
                     SdkFeatureBand manifestFeatureBand;
                     var parts = manifest.Value.Split('/');
+                    
+                    string manifestVersionString = (parts[0]).ToString();
+                    FXVersion _version;
+                    if (!FXVersion.TryParse(manifestVersionString, out _version))
+                    {
+                        if (string.IsNullOrWhiteSpace(manifestVersionString))
+                        {
+                            throw new ArgumentNullException(nameof(manifestVersionString));
+                        }
+                        throw new FormatException($"Error parsing version for workload {manifest.Key}: Invalid version {manifestVersionString}.");
```

Comment:
The string should probably be in a resource so it can get localized as it would get surfaced to the user on the commandline when there's a problem with the rollback file.